pkg/awsclient/client.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsclient
```

Comment:
Thanks for porting this over. Could you get it working with our makefile as well so mockgen is run before final PR?

Reply:
@dgoodwin so i looked at our Makefile and it appears that the 'make generate' target (already a build dependency) handles this already. let me know if you had something else in mind

Reply:
My bad, thanks!